.github/workflows/ci.yaml

```diff
@@ -27,7 +27,19 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python }}
-      - name: Install development dependencies
-        run: pip install -r dev-requirements.txt
-      - name: Build python library
-        run: poetry build
+      - name: Upgrade pip
+        run: |
+          pip install --constraint=./dev-constraints.txt pip
+          pip --version
+      - name: Install poetry
+        run: |
+          pip install --constraint=./dev-constraints.txt poetry
+          poetry --version
+      - name: Install Nox
+        run: |
+          pip install --constraint=./dev-constraints.txt nox nox-poetry
+          nox --version
+      - name: Generate horreum client
+        run: make generate
+      - name: Test horreum
+        run: nox --python=${{ matrix.python }} -s tests
```

Comment:
yeah I agree, but since we are still setting up most of the things I would leave this optimization in a followup PR (I will open an issue to keep track)

Reply:
I concur with trying to avoid "extra work", but I wouldn't regard using common workflows as an optimization -- rather, it's a savings, because you don't have to "reinvent the wheel", and if something needs updating then you can do it in one place, rather than having to track down all the copies of it.